conf/gulp-tasks/templates.gulpfile.js

```diff
@@ -1,145 +1,58 @@
-    },
-    filenameUMD
-  );
-}
-
-function bundleTemplatesIIFE () {
-  return bundleTemplates(
-    {
-      format: 'iife',
-      name: 'TemplateBundle'
-    },
-    filenameIIFE
-  );
-}
-
-function minifyTemplatesUMD (cb) {
-  return src(`./dist/${filenameUMD}`)
-    .pipe(uglify())
-    .pipe(dest('dist'));
-}
-
-function minifyTemplatesIIFE (cb) {
-  return src(`./dist/${filenameIIFE}`)
-    .pipe(uglify())
-    .pipe(dest('dist'));
-}
-
-function cleanFiles () {
-  return del([
-    `./dist/${filenamePrecompiled}`
-  ]);
-}
-
-function watchTemplates (cb) {
+const { parallel, series, watch } = require('gulp');
+const path = require('path');
+const LocalFileParser = require('../i18n/localfileparser');
+const TemplateTaskFactory = require('./template/templatetaskfactory');
+const Translator = require('../i18n/translator');
+const TranslationResolver = require('../i18n/translationresolver');
+
+let devTaskFactory;
+const localFileParser = new LocalFileParser(path.join(__dirname, '../i18n/translations'));
+const devLocale = 'en';
+
+async function createTaskFactory (locale) {
+  const translation = await localFileParser.fetch(locale);
+  const translator = await Translator.create(locale, [], { [locale]: { translation } });
+  const translationResolver = new TranslationResolver(translator, translationResult => translationResult);
+  return new TemplateTaskFactory(translationResolver, locale);
+}
+
+async function watchTemplates () {
+  devTaskFactory = devTaskFactory || await createTaskFactory(devLocale);
   return watch(['./src/ui/templates/**/*.hbs'], {
     ignored: './dist/'
-  }, series(precompileTemplates, bundleTemplatesUMD, cleanFiles));
-}
-
-exports.default = series(
-  precompileTemplates,
-  parallel(
-    series(bundleTemplatesIIFE, minifyTemplatesIIFE),
-    series(bundleTemplatesUMD, minifyTemplatesUMD)
-  ),
-  cleanFiles
-);
-exports.dev = series(precompileTemplates, bundleTemplatesUMD, cleanFiles, watchTemplates);
+  }, series(
+    devTaskFactory.precompileTemplates,
+    devTaskFactory.bundleTemplatesUMD,
+    devTaskFactory.cleanFiles
+  ));
+}
+
+async function devTemplates () {
+  devTaskFactory = devTaskFactory || await createTaskFactory(devLocale);
+  return series(
+    devTaskFactory.precompileTemplates,
+    devTaskFactory.bundleTemplatesUMD,
+    devTaskFactory.cleanFiles,
+    watchTemplates
+  )();
+}
+
+exports.dev = devTemplates;
+
+async function defaultTask () {
+  const locales = ['en'];
+  const localizedTemplateTasks = await Promise.all(locales.map(async locale => {
+    const taskFactory = await createTaskFactory(locale);
+    return series(
+      taskFactory.precompileTemplates,
+      parallel(
+        series(taskFactory.bundleTemplatesIIFE, taskFactory.minifyTemplatesIIFE),
+        series(taskFactory.bundleTemplatesUMD, taskFactory.minifyTemplatesUMD)
+      ),
+      taskFactory.cleanFiles
+    );
+  }));
+  return new Promise(resolve => parallel(...localizedTemplateTasks)(resolve));
```

Comment:
yup that looks right to me!

I wasn't getting proper async completion without passing `resolve` as a parameter to parallel on line 55. I doubt what I have is the best way to write it though, so I'll try to play around a little more and see if there's a way to avoid wrapping it in a promise constructor

Reply:
Ok I think what I have right now is pretty much the best I can come up with